src/ORM/Association/ExternalAssociationTrait.php

```diff
@@ -57,7 +57,7 @@ public function foreignKey($key = null)
     {
         if ($key === null) {
             if ($this->_foreignKey === null) {
-                $this->_foreignKey = $this->_modelKey($this->source()->alias());
+                $this->_foreignKey = $this->_modelKey($this->source()->table());
```

Comment:
@markstory @jadb Wouldn't the foreignkey field be named based on table name? You can have multiple aliases for the table but the foreign key field name is going to be same for all.